pkg/metrics/statsd/statsd.go

```diff
@@ -0,0 +1,152 @@
+package statsd
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/metrics"
+)
+
+// statsd implementation for https://genevamondocs.azurewebsites.net/collect/references/statsdref.html
+
+// Now stubs time.Now
+var Now = time.Now
+
+var (
+	defaultSocket = "/tmp/mdm_statsd.socket"
+)
+
+// Statsd defines internal statsd client
+// It should be cloned, but not modified
+type Statsd struct {
+	account   string
+	namespace string
+	dims      map[string]string
+
+	socket string
+	conn   net.Conn
+	env    env.Interface
+	mu     sync.Mutex
+
+	// for testing
+	Muted bool
```

Comment:
a pkg/metrics/noop which implements the metrics interface but does nothing is preferable.